packages/components/src/color-picker/use-deprecated-props.ts

```diff
@@ -109,39 +107,37 @@ const transformColorStringToLegacyColor = memoize(
 export function useDeprecatedProps(
 	props: LegacyProps | ColorPickerProps
 ): ColorPickerProps {
+	const isUsingLegacy = isLegacyProps( props );
+	const { onChangeComplete } = props as LegacyProps;
+	const { onChange: onChangeProp } = props as ColorPickerProps;
 	const onChange = useCallback(
```

Comment:
To stabilize the `onChange` callback in any meaningful way, I feel like we also need to put the `onChangeComplete`/`onChangeProp` props in a `ref`.

Reply:
This a great thing to consider! I took another look and it seems to me that `onChange` only needs to be as stable as the prop `onChange`/`onChangeComplete`. I.e. Avoid destabilizing it if it's stable. So I think `useCallback` serves well here. Though I did decide could be rewritten a little nicer/simpler and did so in f4aa7c4.

I hope that makes sense. Also I think we'd complicate things trying to stabilize it via a ref because the component would also have to expect a ref as the prop (or else it'd just get the value of the ref and it'd be no different than it is now).

Reply:
Yeah sorry, you're right. With all these exhaustive-deps reviews, my mind was stuck in the mode of "maintaining useEffect behaviors that previous only had partial deps arrays". This one here is not the case at all — it's not even in an effect 🙃

The simplified version is great nonetheless 🙌